concent_api/core/models.py

```diff
@@ -684,3 +686,38 @@ class PendingEthereumTransaction(Model):
     s = DecimalField(max_digits=78, decimal_places=0)
 
     created_at = DateTimeField(auto_now_add=True)
+
+
+class DepositAccount(Model):
+    client = ForeignKey(Client)
+    ethereum_public_key = CharField(max_length=ETHEREUM_ADDRESS_LENGTH)
+    created_at = DateTimeField(auto_now_add=True)
+
+
+class DepositClaim(Model):
+    subtask = ForeignKey(Subtask, blank=True, null=True)
+    payer_deposit_account = ForeignKey(DepositAccount)
+    payee_ethereum_public_key = CharField(max_length=ETHEREUM_ADDRESS_LENGTH)
+    concent_use_case = IntegerField()
+    amount = DecimalField(max_digits=78, decimal_places=0)
+    tx_hash = CharField(max_length=64, blank=True, null=True, unique=True)
+    created_at = DateTimeField(auto_now_add=True)
+    modified_at = DateTimeField(auto_now=True)
+
+    def clean(self) -> None:
+        if self.subtask is None and self.concent_use_case != ConcentUseCase.ForcedPayment.name:
+            raise ValidationError({
+                'subtask: Can be NULL if and only if concent_use_case is ForcedPayment'
+            })
+        if not isinstance(DepositAccount, object) or self.payer_deposit_account.ethereum_public_key == self.payee_ethereum_public_key:
+            raise ValidationError({
+                'payer_deposit_account': 'payer_deposit_account.ethereum_public_key cannot be the same as payee_ethereum_public_key'
+            })
+        if not isinstance(self.payee_ethereum_public_key, str) or self.payee_ethereum_public_key == self.payer_deposit_account.ethereum_public_key:
+            raise ValidationError({
+                'payee_ethereum_public_key': 'Address of the Ethereum account belonging to the entity (requestor, provider or Concent) who is supposed to receive the claim. Cannot be the same as payer_deposit_account.ethereum_public_key'
+            })
+        if not isinstance(self.amount, Decimal) or self.amount <= 0:
```

Comment:
`float` or `Decimal`? The check is about one type and message in exception about another.

Reply:
Decimal.

Reply:
So the error message should be updated.